lib/reek/context/statement_counter.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../ast/node'
+require 'private_attr/everywhere'
+
+module Reek
+  module Context
+    # Responsible for counting the statements in a `CodeContext`.
+    class StatementCounter
+      attr_reader :value
+      private_attr_writer :value
+
+      def initialize
+        @value = 0
+      end
+
+      def increase_by(sexp)
+        return unless sexp
+        case sexp
+        when Reek::AST::Node
+          self.value = value + 1
+        when Array
+          self.value = value + sexp.length
+        else
+          raise ArgumentError, "Invalid type #{sexp} given"
+        end
+      end
```

Comment:
Aaah, right, I actually assumed that private `self.value` would blow up, but then tried it and was surprised it works – turns out it works only on Ruby 2.2.

As for the smell, it seems to show up only once the code is changed to `+=` and `-=`. Strange, but irrelevant for this one – so I’ll merge it. :)